keps/sig-node/4817-resource-claim-device-status/README.md

```diff
@@ -0,0 +1,714 @@
+`ResourceClaim.Status` allowing the network service controller(s) to access
+them.
+
+#### Story 2 - Network Device Status for Troubleshooting
+
+As a Network Administrator, troubleshooting networking issues can be complex
+and time consuming especially when the device characteristics and operational
+status are not readily accessible. The `Devices` field in the
+`ResourceClaim.Status` provides access to comprehensive details regarding
+network interfaces helping to quickly and efficiently identify the issues such
+as error messages on failed network interface configuration, incorrect IP
+assignments or misconfigured network interfaces.
+
+### Notes/Constraints/Caveats (Optional)
+
+The content of `Data` is driver specific and not standardized as part of
+DRA, the interpretation of this field may then vary between controllers and
+users reading it. 
+
+The accuracy of the information depends on the implementation of the DRA
+Drivers, the reported status of the device may not always reflect the real time
+changes of the state of the device.
+
+### Risks and Mitigations
+
+As stated, 3rd party DRA drivers will set and update the `Devices` for
+the device they manage. An access control must be set in place to restrict the
+write access to the appropriate driver (A device status can only be updated by
+the entities that have a direct control over the device(s) being reported).
+
+Adding `Data` as an arbitrary data slice may introduce extra processing
+and storage overhead which might impact performance in a cluster with many
+devices and frequent status updates. In large-scale clusters where many devices
+are allocated, this impact must be considered.
+
+## Design Details
+
+### API
+
+The `ResourceClaimStatus` struct in `pkg/apis/resource/types.go` will be
+extended to include the slice of `Devices`.
+
+`ResourceClaim` validation of the status in
+`pkg/apis/resource/validation/validation.go` will be covered to allow a device
+to be reported only once in the slice, a device is being identified  by
+`<driver name>/<pool name>/<device name>`.
+
+### Write Permission
+
+To prevent unauthorized or accidental modifications by entities that do not 
+have access to a particular resource, a `ValidatingAdmissionPolicy` will be 
+created to validate the entities attempting to update the devices in the 
+`ResourceClaim.Status`.
+
+The `ValidatingAdmissionPolicy` will restrict `ResourceClaim.Status.Devices` 
+to be set only during updates, as the object will have first to be created and 
+allocated, then configured inside the pods. It will also restrict the 
+`ResourceClaim.Status.Devices` to be set only for when the `ResourceClaim` is 
+allocated to a node. Additionally, the allocated node where the `ResourceClaim` 
+is assigned will be used to check if the user/entity updating the 
+`ResourceClaim.Status.Devices` is running on the same node.
+
+Here is a `ResourceClaim` allocated on a node. This would only work for now if 
+exactly one node is set:
```

Comment:
What will happen if the node selector is not of the "one node" form?

Reply:
This was in the conversation on slack because it was not possible to check if the node matches the node selector. But I don't think there was any conclusion for this. @aojea @pohly

Reply:
We need to answer it here. The answer is probably "we fail and reject the update". But it could be "that means node-based authorization isn't needed for this claim" and so we allow it. We need to take a stand for the first alpha, at least.

Reply:
It has to fail close. Otherwise we might accidentally grant all node-agents write permission for something that they shouldn't have access to.

Reply:
I added a paragraph about this in the last commit

Reply:
awesome, thanks

Reply:
I'm not sure I follow - this seems very rigid?  Any driver that doesn't follow this exact standard can't be secured?  Or do we end up embedding thdriver name into the VAP, and have one VAP per driver?

Reply:
yes, I don't love this either, but I figured it's OK for alpha. For beta, I would either want to figure out how to have native, pre-loaded VAP + VAP that can test a label selector, or do an admission plugin.

Reply:
I read the VAP thing as optional/recommended, I imagine different drivers will provide different policies depending on the use cases, but this seems to fit in the model existing network device drivers work